File: 485/b.java
import java.io.*;

class main
{
	public static void main(String[] args) throws Exception
	{
		BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
		int cases = Integer.parseInt(reader.readLine());

		int maxX = Integer.MIN_VALUE,
			minX = Integer.MAX_VALUE,
			maxY = Integer.MIN_VALUE,
			minY = Integer.MAX_VALUE;

		for(int i=0;i<cases;i++)
		{
			String[] q= reader.readLine().split(" ");

			int x = Integer.parseInt(q[0]);
			int y = Integer.parseInt(q[1]);

			maxX = Math.max(maxX,x);
			minX = Math.min(minX,x);

			maxY = Math.max(maxY,y);
			minY = Math.min(minY,y);
		}

		int mx = Math.max(maxX-minX,maxY-minY);
		long sol = (long)mx*mx;
		System.out.println(sol);
	}
}

File: 485/b.cpp
#include <iostream>
#include <algorithm>
#include <climits>
#include <cmath>

using namespace std;

int main()
{
	cin.sync_with_stdio(false);
	int mines,
		min_x = INT_MAX,
		min_y = INT_MAX,
		max_x = -INT_MAX,
		max_y = -INT_MAX;

	cin >> mines;
	long long sol = 0;
	while(mines--)
	{
		int x,y;

		cin >> x >> y;

		min_x = min(min_x,x);
		max_x = max(max_x,x);

		min_y = min(min_y,y);
		max_y = max(max_y,y);
	}

	int a = abs(max_y - min_y);
	int b = abs(max_x - min_x);

	int mx = max(a,b);

	sol = (long long)mx*mx;

	cout << sol << endl;
	return 0;
}